server/data/data-jmap/src/test/java/org/apache/james/jmap/api/preview/MessagePreviewStoreContract.java

```diff
@@ -114,6 +118,28 @@ default void storeShouldBeIdempotent() {
             .isEqualTo(PREVIEW_1);
     }
 
+    @Test
+    default void concurrentStoreShouldStoreAllRecords() throws Exception {
```

Comment:
we probably want to demonstrate that even in case of collision, the result is still ok